### ISSUE TITLE
Harmony-Hub Crashing Work Around

### ISSUE THREAD

**Issue by NexGen-3D-Printing** [KraigM/homebridge-harmonyhub#137]
I have tried all the different plugins for the hub and they all end in issues.

My fix that works flawlessly is to in stall the latest beta of Domoticz, then setup your hubs in that with ease, all gui based, and straight forward, even for a rookie like myself, then install the Domoticz Homebridge plugin. <- make sure its the latest beta, as the stable release has some bugs with Harmony Hub

Works perfect, and very responsive, I have 3 hubs, one controls my climate control system, and in Domoticz you can change the icons and type of the activity or function, and it carries through to home bridge, so I have fan, vent, heat and cool controls and icons to match.

I discovered this while looking for something user friendly to set up some z-wave gear to use in Homebridge, tried Home Assistant and OpenHab2, which were both terrible in comparison to Domoticz.

My setup is:

VM Dietpi for Homebridge
VM Win7 for Domoticz and HCATech (Insteon Control)

Have Insteon, Z-wave, Broadlink, Harmony Hub, Wemo and Alexa all working perfectly together, oh and Siri of course.

Hopefully this helps someone here.

Cheers

KFK

**Comment by crzcrz** [comment]
I disagree with your description of OpenHAB2 as "terrible". I gave up on Homebridge long ago after it became apparent that the Harmony hub plugin was abandoned and have used OpenHAB2 ever since, which works very well and never failed me. It does require writing scripts in order to connect sensors to actuators, which is quite a contrast to the plug-and-play approach of Homebridge with Harmony Hub plugin, but this gives a lot of flexibility.

**Comment by NexGen-3D-Printing** [comment]
I suppose terrible is a little strong, I should have said, "not as easy"

I have HA and Openhab running, I found Homebridge easier to use and work with than both, but Domoticz is really good as a back end, and is great for rookies like myself.

HA and openhab would be easier to setup if you could edit the config files directly through the web interface like config ui x

**Comment by oetelaar24** [comment]
Well, I've been struggling with the plugin as well. It used to work great, on the older versions. Been running the dev version (0.3.0-alpha2) for a while... Always seems to disconnect and error out on the "Error: MAX_CLIENTS=6" error... However, after much searching through the code and trying different things, ultimately, I solved the problem by connecting my Harmony Hub to another WiFi access point...

It's a bit weird, I did suspect this was maybe the issue, but at first I refused that it would be this.. But digging through the code got me weary and I started to think I couldn't fix it this way, because there are a lot of libraries in this project :-)

Running Homebridge in debug mode (homebridge -D), I saw messages like:
```
[2018-8-20 23:31:35] [Harmony Hub] client for hub xyz went offline. re-establish.
[2018-8-20 23:31:35] [Harmony Hub] Updated reachability of undefined to false
[2018-8-20 23:31:35] [Harmony Hub] Updated reachability of undefined to false
[2018-8-20 23:31:36] [Harmony Hub] created new client for hub with uuid xyz
[2018-8-20 23:31:36] [Harmony Hub] Updated reachability of undefined to true
```
And they were very very frequent indeed... 

Finally after changing the WiFi connection on the Harmony Hub (see https://support.myharmony.com/en-us/changing-your-hub-wifi-network ) and rerunning homebridge in debug mode, the errors completely disappeared!!
I'm very convinced this solved the problem for me, however I'm still not convinced it is my WiFi AP that is to blame... it is an enterprise grade Cisco, and only a couple of metres from it... (5 at the most, no drywalls between it, just 2 cardboard-like doors)...
Now I've connected it to the 2.4Ghz WiFi network that my fiber-internet-router emits (same LAN, no VLANs, firewalls etc), which is only about 40cm from the Cisco AP hanging on the wooden wall next to it...

I'm going to investigate this further, because I do believe that my AP should just work fine... But I do want to give you guys this find right now, because I think this may be of great help to you too!

**Comment by NexGen-3D-Printing** [comment]
Still running Domoticz, hasn't failed once, when using the Logitech plugin, my Homebridge would just fail on regular basis, since switching, I have not had a single failure.

**Comment by Mikypan81** [comment]
Hmh, I'm only one Wifi Access point in My house....how I can connect harmony to another access point????

**Comment by oetelaar24** [comment]
Hi @Mikypan81,

There are a couple of things you can try:
1. If possible, create a second wifi network on your router/access point, using a simple name/SSID (no special chars in the name) the wifi network i'm using now has a simple SSID (3 letters and 1 number, no spaces), is 2.4 and 5ghz combined and has WPA2-PSK enabled)
2. If you cannot create a second wifi, because your router/ap doesn't support that. Try adjusting the SSID of your current network to something simple and check if the encryption settings are WPA2-PSK.. My Fritzbox router is set to WPA2(CCMP). According to the router the bug connects to the 2.4Ghz band.
3. Borrow a router/ap from a friend and try connecting your hub to that...

Also, @Kung-Foo-Kamel running Domoticz is nice ofcourse, but seeing that I've been able to solve the problem with this homebridge-plugin by fixing a buggy wifi-connection. Maybe it would be worth investigating if your harmony hub isn't also flaky on the wifi.... That could potentially lead to other issues as well.
